pkg/spoke/controllers/finalizercontroller/finalize_controller.go

```diff
@@ -61,12 +67,15 @@ func (m *FinalizeController) sync(ctx context.Context, controllerContext factory
 	if err != nil {
 		return err
 	}
-	return m.syncManifestWork(ctx, manifestWork)
+	return m.syncManifestWork(ctx, manifestWork, func() {
+		controllerContext.Queue().AddAfter(manifestWorkName, RequeueDelay)
+	})
 }
 
-// syncManifestWork ensures that when a manifestwork has been deleted, everything it created is also deleted before removing
-// the finalizer
-func (m *FinalizeController) syncManifestWork(ctx context.Context, originalManifestWork *workapiv1.ManifestWork) error {
+// syncManifestWork ensures that when a manifestwork has been deleted, everything it created is also deleted.
+// Foreground deletion is implemented, which means all ressources created will be deleted and become invisiable
```

Comment:
typo: resources

Also, this isn't exactly accurate:

> and become invisible in the k8s cluster

Deleted but unfinalized stuff is still visible in the control plane - it is only fully gone once it's finalized. So, after this resources will be deleted but may be unfinalized for arbitrary periods.